Sources/AsyncHTTPClient/HTTPClient.swift

```diff
@@ -118,8 +118,14 @@ public class HTTPClient {
     }
 
     deinit {
-        guard case .shutDown = self.state else {
-            preconditionFailure("Client not shut down before the deinit. Please call client.syncShutdown() when no longer needed.")
+        debugOnly {
+            // We want to crash only in debug mode.
+            switch self.state {
+            case .shutDown:
+                break
+            case .shuttingDown, .upAndRunning:
```

Comment:
Is it possible for `.shuttingDown` to ever be valid here? I.e. if using the async shutdown?

Reply:
You are right, when we are shuttingDown, the HTTPClient is in a retain cycle, that should prevent it from ever going out of memory. Do we want to call this out explicitly?

Reply:
Sure, no harm in it.